### PR TITLE
ci: use withTerraformEnv

### PR DESCRIPTION
**Pull request by v1v**
### What
Uses https://github.com/elastic/apm-pipeline-library/pull/1698

### Why

Simplify the CI logic in favour of some reusable components

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -628,7 +630,9 @@ def dotnet(Closure body){
     ./dotnet-install.sh --install-dir "\${DOTNET_ROOT}" -version '6.0.100'
     """)
     withAzureCredentials(path: "${homePath}", credentialsFile: '.credentials.json') {
-      withTerraform(){
+      withTerraformEnv(version: '0.15.3'){
+        sh 'which terraform || true'
+        sh 'terraform --version || true'
```
Comment (v1v) [inline Jenkinsfile:635]: The installation seems to work

image

```suggestion
        sh 'which terraform || true'
        sh 'terraform --version || true'
```

File: Jenkinsfile [hunk 2]
```diff
@@ -365,7 +365,9 @@ pipeline {
                       dir("${BASE_DIR}"){
                         powershell label: 'Install test tools', script: '.ci\\windows\\test-tools.ps1'
                         withAzureCredentials(path: "${HOME}", credentialsFile: '.credentials.json') {
-                          bat label: 'Test & coverage', script: '.ci/windows/test.bat'
+                          withTerraformEnv(version: '0.15.3') {
```
Comment (v1v) [inline Jenkinsfile:368]: Seems to work 

image

File: Jenkinsfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env groovy
 
-@Library('apm@current') _
+@Library('apm@test/with-terraform') _
```
Comment (v1v) [inline Jenkinsfile:3]:
```suggestion
@Library('apm@current') _
```

File: Jenkinsfile (hunk 1 above)
Comment (v1v) [inline Jenkinsfile:635]:
```suggestion
```

**Review by gregkalapos — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

**Comment by v1v** [comment]
Unrelated test error as far as I see, should I merge this?

**Comment by gregkalapos** [comment]
> Unrelated test error as far as I see, should I merge this?

Yes, those 2 are flaky. I'll need to look into those tests - definitely unrelated to the PR.

**Review by cachedout — APPROVED** [review]

**Pull request merged**